### PR TITLE
Integration spec view

### PR DESCRIPTION
**Pull request by Cabraham1**
## Hello code reviewer! 🌤️
In this milestone of the project I did the following:

## Changes made: ⛏️

- [ ] Fix n+1problem by adding the include method

Before eager loading:

posts_controller rb - Blog - Visual Studio CodeC__Users_Chris_Documents_MICROVERSE_MODULE 5_Blog 6_20_2023 12_45_24 AM

After eager loading:

posts_controller rb - Blog - Visual Studio CodeC__Users_Chris_Documents_MICROVERSE_MODULE 5_Blog 6_20_2023 12_46_48 AM

- [ ] Use Capybara to write integration tests for each view in the project: user_index, user_show, post_index, post_show

### CODE REVIEW

File: spec/views/post_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      @comment1 = Comment.create(author: @user, post: @post, text: 'I like this post comment one')
+      @comment2 = Comment.create(author: @user2, post: @post, text: 'I like this post comment two')
+      @comment3 = Comment.create(author: @user2, post: @post, text: 'I like this post comment three')
+      visit user_post_path(@user, @post)
+    end
+    it "renders the post's title" do
+      expect(page).to have_content(@post.title)
+    end
+  end
+end
```
Comment (zieeco) [inline spec/views/post_show_spec.rb:74]: - Hey, great job so far with the test you've added for the post-show page, however, you've only added a test for the post title, leaving the remaining as seen below undone. Please, consider adding the remaining test:

image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,




Great job so far!  👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯
- N+1 problem solved :heavy_check_mark: 
- No linter errors.✔️
- Integrattion test passed. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    it 'displays number of comments' do
+      visit user_post_path(@user1, @post1)
+      expect(page).to have_content('show post for a given user')
+    end
+
+    it 'displays number of likes' do
+      visit user_post_path(@user1, @post1)
+      expect(page).to have_content('show post for a given user')
+    end
```
Comment (codepantha) [inline spec/views/post_show_spec.rb]: - [ ] The tests that assert that you can see how many comments and likes a post has are testing what is expected of you.
You are currently expecting page to `have_content('show post for a given user')`. Please modify these tests 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added integration tests 📌 
✔️ Linters are passing

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @Cabraham1 , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Required changes implemented ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**